src/test/java/com/microsoft/sqlserver/jdbc/SQLServerConnectionTest.java

```diff
@@ -462,6 +468,126 @@ public void testConnectionPoolGetTwice() throws SQLException {
         }
     }
 
+    /**
+     * Test connection properties: CONCAT_NULL_YIELDS_NULL with SQLServerXADataSource for new connection and pooled connection
+     * @throws SQLException
+     */
+    @Test
+    public void testConcatNullYieldsNull() throws SQLException {
+        SQLServerDataSource ds = new SQLServerDataSource();
+        ds.setURL(connectionString);
+        ds.setConcatNullYieldsNull("OFF");
+        int expectedResultForNewConnection = 0;
+        // Server default is CONCAT_NULL_YIELDS_NULL = ON  
+        int expectedResultForPooledConnection = 1;
+
+        String sqlSelect = "SELECT SESSIONPROPERTY('CONCAT_NULL_YIELDS_NULL')";
+
+        try (Connection con = ds.getConnection(); Statement stmt = con.createStatement()) {
+            try (ResultSet rs = stmt.executeQuery(sqlSelect)) {
+                if (rs.next()) {
+                    assertEquals(expectedResultForNewConnection, rs.getInt(1));
+                } else {
+                    assertTrue(false, "Expected row of data was not found.");
+                }
+            }
+        }

```

Comment:
I'd extract this out into a helper that takes a Connection object so you don't have to duplicate this code so much.